messenger.rst

```diff
@@ -529,84 +823,308 @@ you can disable them like this:
-                        <framework:middleware id="App\Middleware\AnotherMiddleware"/>
-                    </framework:bus>
-                </framework:messenger>
-            </framework:config>
+            <services>
+                <service id="App\MessageHandler\SmsNotificationHandler">
+                   <tag name="messenger.message_handler"/>
+                </service>
+            </services>
         </container>
 
     .. code-block:: php
 
-        // config/packages/messenger.php
-        $container->loadFromExtension('framework', [
-            'messenger' => [
-                'buses' => [
-                    'messenger.bus.default' => [
-                        'middleware' => [
-                            'App\Middleware\MyMiddleware',
-                            'App\Middleware\AnotherMiddleware',
-                        ],
-                    ],
-                ],
-            ],
-        ]);
+        // config/services.php
+        use App\MessageHandler\SmsNotificationHandler;
 
-Note that if the service is abstract, a different instance of the service will
-be created per bus.
+        $container->register(SmsNotificationHandler::class)
+            ->addTag('messenger.message_handler');
 
-Using Middleware Factories
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+Handler Subscriber & Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A handler class can handle multiple messages or configure itself by implementing
+:class:`Symfony\\Component\\Messenger\\Handler\\MessageSubscriberInterface`::
+
+    // src/MessageHandler/SmsNotificationHandler.php
+    namespace App\MessageHandler;
 
-Some third-party bundles and libraries provide configurable middleware via
-factories.
+    use App\Message\OtherSmsNotification;
+    use App\Message\SmsNotification;
+    use Symfony\Component\Messenger\Handler\MessageSubscriberInterface;
 
-For instance, the ``messenger.middleware.doctrine_transaction`` is a
-built-in middleware wired automatically when the DoctrineBundle and the Messenger
-component are installed and enabled.
-This middleware can be configured to define the entity manager to use:
+    class SmsNotificationHandler implements MessageSubscriberInterface
+    {
+        public function __invoke(SmsNotification $message)
+        {
+            // ...
+        }
+
+        public function handleOtherSmsNotification(OtherSmsNotification $message)
+        {
+            // ...
+        }
+
+        public static function getHandledMessages(): iterable
+        {
+            // handle this message on __invoke
+            yield SmsNotification::class;
+
+            // also handle this message on handleOtherSmsNotification
+            yield OtherSmsNotification::class => [
+                'method' => 'handleOtherSmsNotification',
+                //'priority' => 0,
+                //'bus' => 'messenger.bus.default',
+            ];
+        }
+    }
+
+.. _messenger-handlers-different-transports:
+
+Sending Handlers to Different Transports
```

Comment:
this title does not make sense to me. you don't send handlers. maybe you meant something like "Binding Handlers to Different Transports"?